x/rewards/keeper/iprpc.go

```diff
@@ -164,7 +164,7 @@ func (k Keeper) distributeIprpcRewards(ctx sdk.Context, iprpcReward types.IprpcR
 			UsedReward = UsedRewardTemp
 
 			// reward the provider
-			_, _, err := k.dualstakingKeeper.RewardProvidersAndDelegators(ctx, providerCU.Provider, specFund.Spec, providerIprpcReward, string(types.IprpcPoolName), false, false, false)
+			_, _, err := k.dualstakingKeeper.RewardProvidersAndDelegators(ctx, providerCU.Provider, specFund.Spec, providerIprpcReward, string(types.IprpcPoolName), false, false, false, "iprpc rewards")
```

Comment:
not good enough we need the spec there, i don't want to parse it as part of a reason

Reply:
done. see https://github.com/lavanet/lava/commit/93ab557e0b96a7d57b3bbf80dc59e744345bb9e5